// ng-zorro-antd/esm2022/color-picker/color-picker.module.mjs
/**
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/LICENSE
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { NgAntdColorPickerModule } from 'ng-antd-color-picker';
import { NzInputModule } from 'ng-zorro-antd/input';
import { NzInputNumberModule } from 'ng-zorro-antd/input-number';
import { NzPopoverModule } from 'ng-zorro-antd/popover';
import { NzSelectModule } from 'ng-zorro-antd/select';
import { NzColorBlockComponent } from './color-block.component';
import { NzColorFormatComponent } from './color-format.component';
import { NzColorPickerComponent } from './color-picker.component';
import * as i0 from "@angular/core";
class NzColorPickerModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.10", ngImport: i0, type: NzColorPickerModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.2.10", ngImport: i0, type: NzColorPickerModule, declarations: [NzColorPickerComponent, NzColorBlockComponent, NzColorFormatComponent], imports: [CommonModule,
            NgAntdColorPickerModule,
            NzPopoverModule,
            NzSelectModule,
            FormsModule,
            NzInputNumberModule,
            NzInputModule,
            ReactiveFormsModule], exports: [NzColorPickerComponent, NzColorBlockComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.2.10", ngImport: i0, type: NzColorPickerModule, imports: [CommonModule,
            NgAntdColorPickerModule,
            NzPopoverModule,
            NzSelectModule,
            FormsModule,
            NzInputNumberModule,
            NzInputModule,
            ReactiveFormsModule] }); }
}
export { NzColorPickerModule };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.10", ngImport: i0, type: NzColorPickerModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [NzColorPickerComponent, NzColorBlockComponent, NzColorFormatComponent],
                    imports: [
                        CommonModule,
                        NgAntdColorPickerModule,
                        NzPopoverModule,
                        NzSelectModule,
                        FormsModule,
                        NzInputNumberModule,
                        NzInputModule,
                        ReactiveFormsModule
                    ],
                    exports: [NzColorPickerComponent, NzColorBlockComponent]
                }]
        }] });
