// ng-zorro-antd/esm2022/alert/alert.module.mjs
/**
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/LICENSE
 */
import { BidiModule } from '@angular/cdk/bidi';
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { NzOutletModule } from 'ng-zorro-antd/core/outlet';
import { NzIconModule } from 'ng-zorro-antd/icon';
import { NzAlertComponent } from './alert.component';
import * as i0 from "@angular/core";
class NzAlertModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.10", ngImport: i0, type: NzAlertModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.2.10", ngImport: i0, type: NzAlertModule, declarations: [NzAlertComponent], imports: [BidiModule, CommonModule, NzIconModule, NzOutletModule], exports: [NzAlertComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.2.10", ngImport: i0, type: NzAlertModule, imports: [BidiModule, CommonModule, NzIconModule, NzOutletModule] }); }
}
export { NzAlertModule };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.10", ngImport: i0, type: NzAlertModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [NzAlertComponent],
                    exports: [NzAlertComponent],
                    imports: [BidiModule, CommonModule, NzIconModule, NzOutletModule]
                }]
        }] });
