// ng-zorro-antd/esm2022/affix/affix.module.mjs
/**
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/LICENSE
 */
import { BidiModule } from '@angular/cdk/bidi';
import { PlatformModule } from '@angular/cdk/platform';
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { NzAffixComponent } from './affix.component';
import * as i0 from "@angular/core";
class NzAffixModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.10", ngImport: i0, type: NzAffixModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.2.10", ngImport: i0, type: NzAffixModule, declarations: [NzAffixComponent], imports: [BidiModule, CommonModule, PlatformModule], exports: [NzAffixComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.2.10", ngImport: i0, type: NzAffixModule, imports: [BidiModule, CommonModule, PlatformModule] }); }
}
export { NzAffixModule };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.10", ngImport: i0, type: NzAffixModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [NzAffixComponent],
                    exports: [NzAffixComponent],
                    imports: [BidiModule, CommonModule, PlatformModule]
                }]
        }] });
