// ng-zorro-antd/esm2022/core/element-patch/element-patch.directive.mjs
/**
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/LICENSE
 */
import { Directive } from '@angular/core';
import * as i0 from "@angular/core";
/**
 * A patch directive to select the element of a component.
 */
class NzElementPatchDirective {
    get nativeElement() {
        return this.elementRef.nativeElement;
    }
    constructor(elementRef) {
        this.elementRef = elementRef;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.10", ngImport: i0, type: NzElementPatchDirective, deps: [{ token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.2.10", type: NzElementPatchDirective, selector: "[nzElement], [nz-element]", exportAs: ["nzElement"], ngImport: i0 }); }
}
export { NzElementPatchDirective };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.10", ngImport: i0, type: NzElementPatchDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[nzElement], [nz-element]',
                    exportAs: 'nzElement'
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }]; } });
