// ng-zorro-antd/esm2022/progress/progress.module.mjs
/**
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/LICENSE
 */
import { BidiModule } from '@angular/cdk/bidi';
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { NzOutletModule } from 'ng-zorro-antd/core/outlet';
import { NzIconModule } from 'ng-zorro-antd/icon';
import { NzProgressComponent } from './progress.component';
import * as i0 from "@angular/core";
class NzProgressModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.10", ngImport: i0, type: NzProgressModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.2.10", ngImport: i0, type: NzProgressModule, declarations: [NzProgressComponent], imports: [BidiModule, CommonModule, NzIconModule, NzOutletModule], exports: [NzProgressComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.2.10", ngImport: i0, type: NzProgressModule, imports: [BidiModule, CommonModule, NzIconModule, NzOutletModule] }); }
}
export { NzProgressModule };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.10", ngImport: i0, type: NzProgressModule, decorators: [{
            type: NgModule,
            args: [{
                    exports: [NzProgressComponent],
                    declarations: [NzProgressComponent],
                    imports: [BidiModule, CommonModule, NzIconModule, NzOutletModule]
                }]
        }] });
