// ng-zorro-antd/esm2022/code-editor/code-editor.module.mjs
/**
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/LICENSE
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { NzIconModule } from 'ng-zorro-antd/icon';
import { NzSpinModule } from 'ng-zorro-antd/spin';
import { NzCodeEditorComponent } from './code-editor.component';
import * as i0 from "@angular/core";
class NzCodeEditorModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.10", ngImport: i0, type: NzCodeEditorModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.2.10", ngImport: i0, type: NzCodeEditorModule, declarations: [NzCodeEditorComponent], imports: [CommonModule, NzIconModule, NzSpinModule], exports: [NzCodeEditorComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.2.10", ngImport: i0, type: NzCodeEditorModule, imports: [CommonModule, NzIconModule, NzSpinModule] }); }
}
export { NzCodeEditorModule };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.10", ngImport: i0, type: NzCodeEditorModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [NzCodeEditorComponent],
                    imports: [CommonModule, NzIconModule, NzSpinModule],
                    exports: [NzCodeEditorComponent]
                }]
        }] });
