// ng-zorro-antd/esm2022/input-number/input-number.module.mjs
/**
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/LICENSE
 */
import { BidiModule } from '@angular/cdk/bidi';
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { NzFormPatchModule } from 'ng-zorro-antd/core/form';
import { NzOutletModule } from 'ng-zorro-antd/core/outlet';
import { NzIconModule } from 'ng-zorro-antd/icon';
import { NzInputNumberGroupSlotComponent } from './input-number-group-slot.component';
import { NzInputNumberGroupComponent, NzInputNumberGroupWhitSuffixOrPrefixDirective } from './input-number-group.component';
import { NzInputNumberComponent } from './input-number.component';
import * as i0 from "@angular/core";
class NzInputNumberModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.10", ngImport: i0, type: NzInputNumberModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.2.10", ngImport: i0, type: NzInputNumberModule, declarations: [NzInputNumberComponent,
            NzInputNumberGroupComponent,
            NzInputNumberGroupWhitSuffixOrPrefixDirective,
            NzInputNumberGroupSlotComponent], imports: [BidiModule, CommonModule, FormsModule, NzOutletModule, NzIconModule, NzFormPatchModule], exports: [NzInputNumberComponent, NzInputNumberGroupComponent, NzInputNumberGroupWhitSuffixOrPrefixDirective] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.2.10", ngImport: i0, type: NzInputNumberModule, imports: [BidiModule, CommonModule, FormsModule, NzOutletModule, NzIconModule, NzFormPatchModule] }); }
}
export { NzInputNumberModule };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.10", ngImport: i0, type: NzInputNumberModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [BidiModule, CommonModule, FormsModule, NzOutletModule, NzIconModule, NzFormPatchModule],
                    declarations: [
                        NzInputNumberComponent,
                        NzInputNumberGroupComponent,
                        NzInputNumberGroupWhitSuffixOrPrefixDirective,
                        NzInputNumberGroupSlotComponent
                    ],
                    exports: [NzInputNumberComponent, NzInputNumberGroupComponent, NzInputNumberGroupWhitSuffixOrPrefixDirective]
                }]
        }] });
