// ng-zorro-antd/esm2022/comment/comment.module.mjs
/**
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/LICENSE
 */
import { BidiModule } from '@angular/cdk/bidi';
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { NzOutletModule } from 'ng-zorro-antd/core/outlet';
import { NzCommentActionComponent, NzCommentActionHostDirective, NzCommentAvatarDirective, NzCommentContentDirective } from './comment-cells';
import { NzCommentComponent } from './comment.component';
import * as i0 from "@angular/core";
const NZ_COMMENT_CELLS = [
    NzCommentAvatarDirective,
    NzCommentContentDirective,
    NzCommentActionComponent,
    NzCommentActionHostDirective
];
class NzCommentModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.10", ngImport: i0, type: NzCommentModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.2.10", ngImport: i0, type: NzCommentModule, declarations: [NzCommentComponent, NzCommentAvatarDirective,
            NzCommentContentDirective,
            NzCommentActionComponent,
            NzCommentActionHostDirective], imports: [BidiModule, CommonModule, NzOutletModule], exports: [NzCommentComponent, NzCommentAvatarDirective,
            NzCommentContentDirective,
            NzCommentActionComponent,
            NzCommentActionHostDirective] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.2.10", ngImport: i0, type: NzCommentModule, imports: [BidiModule, CommonModule, NzOutletModule] }); }
}
export { NzCommentModule };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.10", ngImport: i0, type: NzCommentModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [BidiModule, CommonModule, NzOutletModule],
                    exports: [NzCommentComponent, ...NZ_COMMENT_CELLS],
                    declarations: [NzCommentComponent, ...NZ_COMMENT_CELLS]
                }]
        }] });
