// ng-zorro-antd/esm2022/spin/spin.module.mjs
/**
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/LICENSE
 */
import { BidiModule } from '@angular/cdk/bidi';
import { ObserversModule } from '@angular/cdk/observers';
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { NzSpinComponent } from './spin.component';
import * as i0 from "@angular/core";
class NzSpinModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.10", ngImport: i0, type: NzSpinModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.2.10", ngImport: i0, type: NzSpinModule, declarations: [NzSpinComponent], imports: [BidiModule, CommonModule, ObserversModule], exports: [NzSpinComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.2.10", ngImport: i0, type: NzSpinModule, imports: [BidiModule, CommonModule, ObserversModule] }); }
}
export { NzSpinModule };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.10", ngImport: i0, type: NzSpinModule, decorators: [{
            type: NgModule,
            args: [{
                    exports: [NzSpinComponent],
                    declarations: [NzSpinComponent],
                    imports: [BidiModule, CommonModule, ObserversModule]
                }]
        }] });
