// ng-zorro-antd/esm2022/experimental/image/image.module.mjs
/**
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/LICENSE
 */
import { BidiModule } from '@angular/cdk/bidi';
import { DragDropModule } from '@angular/cdk/drag-drop';
import { OverlayModule } from '@angular/cdk/overlay';
import { PlatformModule } from '@angular/cdk/platform';
import { PortalModule } from '@angular/cdk/portal';
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { NzIconModule } from 'ng-zorro-antd/icon';
import { NzImageModule as ImageModule } from 'ng-zorro-antd/image';
import { NzPipesModule } from 'ng-zorro-antd/pipes';
import { NzImageViewComponent } from './image.component';
import * as i0 from "@angular/core";
class NzImageModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.10", ngImport: i0, type: NzImageModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.2.10", ngImport: i0, type: NzImageModule, declarations: [NzImageViewComponent], imports: [BidiModule,
            OverlayModule,
            PortalModule,
            DragDropModule,
            CommonModule,
            NzIconModule,
            NzPipesModule,
            PlatformModule,
            ImageModule], exports: [NzImageViewComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.2.10", ngImport: i0, type: NzImageModule, imports: [BidiModule,
            OverlayModule,
            PortalModule,
            DragDropModule,
            CommonModule,
            NzIconModule,
            NzPipesModule,
            PlatformModule,
            ImageModule] }); }
}
export { NzImageModule };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.10", ngImport: i0, type: NzImageModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        BidiModule,
                        OverlayModule,
                        PortalModule,
                        DragDropModule,
                        CommonModule,
                        NzIconModule,
                        NzPipesModule,
                        PlatformModule,
                        ImageModule
                    ],
                    exports: [NzImageViewComponent],
                    declarations: [NzImageViewComponent]
                }]
        }] });
