// ng-zorro-antd/esm2022/skeleton/skeleton.module.mjs
/**
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/LICENSE
 */
import { BidiModule } from '@angular/cdk/bidi';
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { NzSkeletonElementAvatarComponent, NzSkeletonElementButtonComponent, NzSkeletonElementDirective, NzSkeletonElementImageComponent, NzSkeletonElementInputComponent } from './skeleton-element.component';
import { NzSkeletonComponent } from './skeleton.component';
import * as i0 from "@angular/core";
class NzSkeletonModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.10", ngImport: i0, type: NzSkeletonModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.2.10", ngImport: i0, type: NzSkeletonModule, declarations: [NzSkeletonComponent,
            NzSkeletonElementDirective,
            NzSkeletonElementButtonComponent,
            NzSkeletonElementAvatarComponent,
            NzSkeletonElementImageComponent,
            NzSkeletonElementInputComponent], imports: [BidiModule, CommonModule], exports: [NzSkeletonComponent,
            NzSkeletonElementDirective,
            NzSkeletonElementButtonComponent,
            NzSkeletonElementAvatarComponent,
            NzSkeletonElementImageComponent,
            NzSkeletonElementInputComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.2.10", ngImport: i0, type: NzSkeletonModule, imports: [BidiModule, CommonModule] }); }
}
export { NzSkeletonModule };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.10", ngImport: i0, type: NzSkeletonModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        NzSkeletonComponent,
                        NzSkeletonElementDirective,
                        NzSkeletonElementButtonComponent,
                        NzSkeletonElementAvatarComponent,
                        NzSkeletonElementImageComponent,
                        NzSkeletonElementInputComponent
                    ],
                    imports: [BidiModule, CommonModule],
                    exports: [
                        NzSkeletonComponent,
                        NzSkeletonElementDirective,
                        NzSkeletonElementButtonComponent,
                        NzSkeletonElementAvatarComponent,
                        NzSkeletonElementImageComponent,
                        NzSkeletonElementInputComponent
                    ]
                }]
        }] });
