// ng-zorro-antd/esm2022/qr-code/qrcode.module.mjs
/**
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/LICENSE
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { NzButtonModule } from 'ng-zorro-antd/button';
import { NzIconModule } from 'ng-zorro-antd/icon';
import { NzSpinModule } from 'ng-zorro-antd/spin';
import { NzQRCodeComponent } from './qrcode.component';
import * as i0 from "@angular/core";
class NzQRCodeModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.10", ngImport: i0, type: NzQRCodeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.2.10", ngImport: i0, type: NzQRCodeModule, declarations: [NzQRCodeComponent], imports: [NzSpinModule, CommonModule, NzButtonModule, NzIconModule], exports: [NzQRCodeComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.2.10", ngImport: i0, type: NzQRCodeModule, imports: [NzSpinModule, CommonModule, NzButtonModule, NzIconModule] }); }
}
export { NzQRCodeModule };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.10", ngImport: i0, type: NzQRCodeModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [NzQRCodeComponent],
                    exports: [NzQRCodeComponent],
                    imports: [NzSpinModule, CommonModule, NzButtonModule, NzIconModule]
                }]
        }] });
