// ng-zorro-antd/esm2022/back-top/back-top.module.mjs
/**
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/LICENSE
 */
import { BidiModule } from '@angular/cdk/bidi';
import { PlatformModule } from '@angular/cdk/platform';
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { NzIconModule } from 'ng-zorro-antd/icon';
import { NzBackTopComponent } from './back-top.component';
import * as i0 from "@angular/core";
class NzBackTopModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.10", ngImport: i0, type: NzBackTopModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.2.10", ngImport: i0, type: NzBackTopModule, declarations: [NzBackTopComponent], imports: [BidiModule, CommonModule, PlatformModule, NzIconModule], exports: [NzBackTopComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.2.10", ngImport: i0, type: NzBackTopModule, imports: [BidiModule, CommonModule, PlatformModule, NzIconModule] }); }
}
export { NzBackTopModule };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.10", ngImport: i0, type: NzBackTopModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [NzBackTopComponent],
                    exports: [NzBackTopComponent],
                    imports: [BidiModule, CommonModule, PlatformModule, NzIconModule]
                }]
        }] });
