// ng-zorro-antd/esm2022/space/space.module.mjs
/**
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/LICENSE
 */
import { BidiModule } from '@angular/cdk/bidi';
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { NzSpaceItemDirective } from './space-item.directive';
import { NzSpaceComponent } from './space.component';
import * as i0 from "@angular/core";
class NzSpaceModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.10", ngImport: i0, type: NzSpaceModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.2.10", ngImport: i0, type: NzSpaceModule, declarations: [NzSpaceComponent, NzSpaceItemDirective], imports: [BidiModule, CommonModule], exports: [NzSpaceComponent, NzSpaceItemDirective] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.2.10", ngImport: i0, type: NzSpaceModule, imports: [BidiModule, CommonModule] }); }
}
export { NzSpaceModule };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.10", ngImport: i0, type: NzSpaceModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [NzSpaceComponent, NzSpaceItemDirective],
                    exports: [NzSpaceComponent, NzSpaceItemDirective],
                    imports: [BidiModule, CommonModule]
                }]
        }] });
